### PR TITLE
fix(tests): send Origin headers with resource requests

### PR DESCRIPTION
**Pull request by jrgm**
I'm not sure this makes sense, but it is more like a real browser request, and in stage a change to using S3 as the origin (which will have to be done in latest too) means that the test/server/routes.js test fail without this change.

r? - @shane-tomlinson and @vladikoff

### CODE REVIEW

**Comment by jrgm** [comment]
> and in stage a change to using S3 as the origin

To clarify, s3 as the origin for CDN requests for static content, which will not unilaterally add and 'Access-Control-Allow-Origin' in the response, but will send one if 'Origin' is on the request.

**Comment by jbuck** [comment]
I wonder if we want to do some conditional sending of the Origin header based on isExternalUrl() like how the browser does it.

**Comment by jrgm** [comment]
> I wonder if we want to do some conditional sending of the Origin header based on isExternalUrl() like how the browser does it.

Yeah, probably.

**Comment by shane-tomlinson** [comment]
@jrgm, @jbuck - I'm a tad bit confused, are we going to start deploying static content to S3, where the CDN heads will use as the origin instead of hitting nginx/node?

**Comment by jbuck** [comment]
@shane-tomlinson that's exactly what we're doing

File: tests/server/routes.js [hunk 1]
```diff
@@ -214,13 +216,19 @@ define([
...
 
-  function checkUrls(urls) {
+  function checkUrls(origin, urls) {
     var requests = urls.map(function (url) {
       if (checkedUrlPromises[url]) {
         return checkedUrlPromises[url];
       }
 
-      var promise = makeRequest(url, {})
+      var requestOptions = {
```
Comment (shane-tomlinson) [inline tests/server/routes.js]: My comment is the same as @jbuck's, perhaps we should call `isExternalSite` and only add the `Origin` header if making a request to a 3rd party.

**Comment by shane-tomlinson** [comment]
Back to @jrgm - other than the one comment @jbuck pointed out to only add Origin if making a request to a 3rd party site to better approximate browser behavior, LGTM. Back to you!

**Comment by jrgm** [comment]
Updated to use `doesURLRequireCORS(url)` to determine if `Origin:` should be added.

**Pull request merged**